### ISSUE TITLE
Scan amp-story for media elements to determine MediaPool size

### ISSUE THREAD

**Issue by newmuis** [ampproject/amphtml#13254]
Right now, we always create a `MediaPool` with 8 videos and 4 audio elements.  We should, instead, scan at build time for how many elements a story will actually use, keeping the existing numbers as maxima.

**Comment by newmuis** [comment]
To do this, we will need to initialize `this.mediaPool_` in the `buildCallback()` of `AmpStory` instead of the constructor.  Then, we can change the implementation of `getMaxMediaElementCounts()` in `AmpStory`.

From:

https://github.com/ampproject/amphtml/blob/376d3044961b95a2424868a25ced3786cbcc91cb/extensions/amp-story/0.1/amp-story.js#L1324-L1327

To:

```javascript
/** @override */
getMaxMediaElementCounts() {
  return {
    [MediaType.AUDIO]: Math.min(
        scopedQuerySelectorAll(this.element, 'amp-audio, [background-audio]'),
        MAX_MEDIA_ELEMENT_COUNTS[MediaType.AUDIO]),
    [MediaType.VIDEO]: Math.min(
        scopedQuerySelectorAll(this.element, 'amp-video'),
        MAX_MEDIA_ELEMENT_COUNTS[MediaType.VIDEO]),
  };
}
```

**Comment by newmuis** [comment]
Fixed by #13590
